rustls-libssl/src/cache.rs

```diff
@@ -39,6 +54,8 @@ impl SessionCaches {
     pub fn set_size(&mut self, size: usize) {
         self.max_size = size;
         self.server.set_size(size);
+        // divergence: openssl can change the size without emptying the (client) cache
+        self.client.take();
```

Comment:
Not sure about that. If someone went from a size of 10 to 0 this would leave them with a very small cache, albeit one that still had the old items in.